diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4259688..51e05d7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -31,7 +31,7 @@ find_package(fhiclcpp REQUIRED EXPORT)
 find_package(cetlib REQUIRED EXPORT)
 find_package(cetlib_except REQUIRED EXPORT)
 
-find_package(Boost COMPONENTS unit_test_framework REQUIRED EXPORT)
+find_package(Boost COMPONENTS unit_test_framework filesystem REQUIRED EXPORT)
 find_package(CLHEP COMPONENTS Geometry Vector REQUIRED EXPORT)
 find_package(ROOT COMPONENTS Core GenVector Geom MathCore Matrix Physics REQUIRED EXPORT)
 
diff --git a/larcorealg/CoreUtils/makeValueIndex.h b/larcorealg/CoreUtils/makeValueIndex.h
index 2074776..8f4a15a 100644
--- a/larcorealg/CoreUtils/makeValueIndex.h
+++ b/larcorealg/CoreUtils/makeValueIndex.h
@@ -60,9 +60,12 @@ namespace util {
   template <typename Coll>
   auto makeValueIndex(Coll const& coll)
   {
+#if (__cplusplus < 202000L)
     return makeValueIndex(coll, util::pre_std::identity());
+#else
+    return makeValueIndex(coll, std::identity());
+#endif
   }
-
 } // namespace util
 
 //------------------------------------------------------------------------------
